tests/ui/use_self.rs

```diff
@@ -312,17 +312,18 @@ mod issue4140 {
         fn try_from(value: T) -> Result<Self, Error<Self::From, Self::To>>;
     }
 
-    impl<F, T> TryFrom<F> for T
-    where
-        T: From<F>,
-    {
-        type From = T::From;
-        type To = T::To;
-
-        fn try_from(value: F) -> Result<Self, Error<Self::From, Self::To>> {
-            Ok(From::from(value))
-        }
-    }
+    // FIXME: Suggested fix results in infinite recursion.
```

Comment:
This should be fine. This test only exists to show, that our suggestion is bad, so keeping it commented out also makes the same point.